src/content/docs/node/introduction.mdx

```diff
@@ -1,3 +1,502 @@
+
+CommonJS uses `module.exports` and `require` for exporting and importing modules, while ES Modules use `export` and `import`.
+
+| Feature        | CommonJS                          | ES Modules                                                  |
+| -------------- | --------------------------------- | ----------------------------------------------------------- |
+| Syntax         | `module.exports` / `require`      | `export` / `import`                                         |
+| File Extension | `.js`                             | `.mjs` (or `.js` with `"type": "module"` in `package.json`) |
+| Import Type    | Synchronous                       | Asynchronous                                                |
+| Default Usage  | Older Node.js apps, simpler setup | Modern apps, full browser support                           |
+
+##### Example: CommonJS
+
+```js
+// math.js
+module.exports = { add: (a, b) => a + b };
+
+// app.js
+const math = require("./math");
+console.log(math.add(2, 3));
+```
+
+##### Example: ES Modules
+
+```js
+// math.mjs
+export function add(a, b) {
+  return a + b;
+}
+
+// app.mjs
+import { add } from "./math.mjs";
+console.log(add(2, 3));
+```
+
+### Built-in Modules in Node.js
+
+Node.js provides several built-in modules that are essential for building applications. These modules cover common tasks like file handling, networking, and event handling. Here are a few important ones:
+
+##### 1. File System (`fs`)
+
+The `fs` module allows you to interact with the file system, enabling you to read, write, update, and delete files.
+
+```js
+// Import the 'fs' module
+const fs = require("fs");
+
+fs.readFile("example.txt", "utf8", (err, data) => {
+  if (err) throw err;
+  console.log(data);
+});
+```
+
+##### 1. File System (`fs`)
+
+The `fs` module allows you to interact with the file system, enabling you to read, write, update, and delete files.
+
+**Example: Reading a file**
+
+```js
+// Import the 'fs' module
+const fs = require("fs");
+
+fs.readFile("example.txt", "utf8", (err, data) => {
+  if (err) throw err;
+  console.log(data);
+});
+```
+
+##### 2. HTTP
+
+The `http` module enables the creation of web servers and handling of HTTP requests.
+**Example: Simple HTTP server**
+
+```js
+const http = require("http");
+
+http
+  .createServer((req, res) => {
+    res.writeHead(200, { "Content-Type": "text/plain" });
+    res.end("Hello, World!\n");
+  })
+  .listen(3000, () => {
+    console.log("Server running at http://localhost:3000/");
+  });
+```
+
+##### 3. Events
+
+The `events` module provides a way to work with event-driven programming. Node.js applications frequently rely on event emitters to trigger and handle asynchronous actions.
+
+**Example: Using EventEmitter**
+
+```js
+const EventEmitter = require("events");
+const eventEmitter = new EventEmitter();
+
+// Event listener
+eventEmitter.on("greet", () => {
+  console.log("Hello, World!");
+});
+
+// Emit event
+eventEmitter.emit("greet");
+```
```

Comment:
After this please add the one more heading called "References" & add 4-5 points with links for the best documentations regarding the NodeJs & it's related topics.